### PR TITLE
Add create release for github after deploy

### PR DESCRIPTION
**Pull request by abtris**
Try make implementation #45 can you look if this can work?

### CODE REVIEW

File: blackbelt/handle_github.py [hunk 1]
```diff
@@ -270,6 +270,30 @@ def deploy(pr_url):
...
+def create_release(ref, payload, description):
+    """ Create release in github after deploy to production """
+
+    url = "https://api.github.com/repos/%(owner)s/%(name)s/deployments"
+
+    payload = {
+        'ref': ref,
+        'payload': payload,
+        'description': description
+    }
```
Comment (LilyJ) [inline blackbelt/handle_github.py]: @abtris I can see the redundancy in variable name i.e. payload. You might want rename one or another to something else.

**Comment by LilyJ** [comment]
leaving this to @Almad

File: blackbelt/handle_github.py (hunk 1 above)
Comment (abtris) [inline blackbelt/handle_github.py]: good point!

File: blackbelt/handle_github.py [hunk 2]
```diff
@@ -270,6 +270,30 @@ def deploy(pr_url):
...
         move_to_deployed(card_id=ticket_id, comment=comment)
+        create_release(ref=merge_info['sha'], payload='', description="Deployed to production")
     except ValueError:
         if click.prompt("Moving card failed. Open PR in browser?", default=True):
             webbrowser.open(merge_info['html_url'])
+
+def create_release(ref, payload, description):
+    """ Create release in github after deploy to production """
+
+    url = "https://api.github.com/repos/%(owner)s/%(name)s/deployments"
```
Comment (honzajavorek) [inline blackbelt/handle_github.py]: how do you get the placeholders interpolated into values? i don't see it anywhere

Comment (abtris) [inline blackbelt/handle_github.py]: Good point! Missing

File: blackbelt/handle_github.py [hunk 3]
```diff
@@ -270,6 +270,30 @@ def deploy(pr_url):
...
+    url = "https://api.github.com/repos/%(owner)s/%(name)s/deployments" % repo_info
+
+    body = {
+        'ref': ref,
+        'payload': payload,
+        'description': description
+    }
+
+    headers = {
+        'Authorization': "token %s" % config['github']['access_token']
```
Comment (kylef) [inline blackbelt/handle_github.py]: Since this is sending a JSON body, should this include the `Content-Type: application/json` header?

Comment (abtris) [inline blackbelt/handle_github.py]: @Almad It's necessary or python requests library making this automatically. I don't see this header   in this file.

Comment (Almad) [inline blackbelt/handle_github.py]: @kylef @abtris it will be filled automatically if you'd use `json=body` instead of `data=json.dumps(body)`

**Comment by XVincentX** [comment]
@Almad @abtris 
What are we waiting for?

**Comment by abtris** [comment]
@Almad It's ready to merge?

**Comment by Almad** [comment]
Hoped to get the json header/nitpick pointed by @kylef fixed, but OK—relying that you've tested it.

File: blackbelt/handle_github.py [hunk 4]
```diff
@@ -270,6 +270,30 @@ def deploy(pr_url):
     try:
         ticket_id = get_pr_ticket_id(merge_info['description'])
         move_to_deployed(card_id=ticket_id, comment=comment)
+        create_release(ref=merge_info['sha'], payload='', description="Deployed to production")
```
Comment (XVincentX) [inline blackbelt/handle_github.py]: Probably this **try** contains too many istructions. A **create_release** error will always show a moving failed error.

Comment (Almad) [inline blackbelt/handle_github.py]: Agreed, it should have its own try (and doesn't make sense to open in browser if it fais.

Comment (XVincentX) [inline blackbelt/handle_github.py]: I'll take care this night.

**Pull request merged**